src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -81,7 +54,7 @@
  * @author Justin Deoliveira, OpenGeo
  */
 @SuppressWarnings("unused")
-public class GeoServerDataDirectory implements ResourceStore {
+public class GeoServerDataDirectory {
```

Comment:
I am really excited you were able to do this, excellent work @NielsCharlier for making this possible.

It is an API change, but I guess that is what we are up to here, trying to factor out things correctly so code can consistently get at a ResourceStore.